### DISCUSSION TITLE
Incompressible simulation of turbomachine

### DISCUSSION THREAD

**Discussion by z-g-h** [su2code/SU2#1665] [Q&A]
hello,
when I simulate an axial machine which is incompressible flow.
however i get the Segmentation fault.
`---------------------- Turbomachinery Preprocessing ---------------------

Initialize Turbo Vertex Structure.
Number of span-wise sections in Zone 0: 40.
TURBOMACHINERY folder creation failed.
Number of span-wise sections in Zone 1: 33.
Max number of span-wise sections among all zones: 40.
Initialize solver containers for average and performance quantities.
Compute inflow and outflow average geometric quantities.
Set span-wise sections between zones on Mixing-Plane interface.
Transfer average geometric quantities to zone 0.
Inlet area for Row 1: 20.9591 cm^2.
Oulet area for Row 1: 16.0449 cm^2.
Recomputed number of blades for Row 1: -32.
Inlet area for Row 2: 39.4948 cm^2.
Oulet area for Row 2: 49.3475 cm^2.
Recomputed number of blades for Row 2: 13.
Preprocessing of the Mixing-Plane Interface.
Initialize turbomachinery solution quantities.
Initialize inflow and outflow average solution quantities.
Segmentation fault
`
this is my config file:
% Enable multizone feature
MULTIZONE= YES
%:q

% List of config files for zone-specific options
CONFIG_LIST=(D.cfg, B.cfg)
%
% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
%
% Physical governing equations (EULER, NAVIER_STOKES,
%                               WAVE_EQUATION, HEAT_EQUATION, LINEAR_ELASTICITY,
%                               POISSON_EQUATION)                         
SOLVER= INC_EULER
%
% Specify turbulent model (NONE, SA, SST)
KIND_TURB_MODEL= SST
%
% Mathematical problem (DIRECT, ADJOINT, LINEARIZED)
MATH_PROBLEM= DIRECT
%
% Restart solution (NO, YES)
RESTART_SOL= NO
%
%

% ------------------------- UNSTEADY SIMULATION -------------------------------%
%
% Unsteady simulation (NO, TIME_STEPPING, DUAL_TIME_STEPPING-1ST_ORDER, 
%                      DUAL_TIME_STEPPING-2ND_ORDER)
TIME_MARCHING= NO
% ---------------- INCOMPRESSIBLE FLOW CONDITION DEFINITION -------------------%
% Solve the energy equation in the incompressible flow solver
INC_ENERGY_EQUATION = NO
%
% Initial density for incompressible flows
% (1.2886 kg/m^3 by default (air), 998.2 Kg/m^3 (water))
INC_DENSITY_INIT=  998.2
%
% Initial velocity for incompressible flows (1.0,0,0 m/s by default)
INC_VELOCITY_INIT= ( 5.0, 0.0, 0.0 )

% ---------------------- REFERENCE VALUE DEFINITION ---------------------------%
%
% Reference origin for moment computation
REF_ORIGIN_MOMENT_X = 0.00
REF_ORIGIN_MOMENT_Y = 0.00
REF_ORIGIN_MOMENT_Z = 0.00
%
% Reference length for pitching, rolling, and yawing non-dimensional moment
REF_LENGTH= 1.0
%
% Reference area for force coefficients (0 implies automatic calculation)
REF_AREA= 1.0
%
%
% Flow non-dimensionalization (DIMENSIONAL, FREESTREAM_PRESS_EQ_ONE,
%                              FREESTREAM_VEL_EQ_MACH, FREESTREAM_VEL_EQ_ONE)
REF_DIMENSIONALIZATION= DIMENSIONAL
%
% --------------------------- VISCOSITY MODEL ---------------------------------%
%
% Viscosity model (SUTHERLAND, CONSTANT_VISCOSITY).
VISCOSITY_MODEL= CONSTANT_VISCOSITY
%
% Molecular Viscosity that would be constant (1.716E-5 by default)
MU_CONSTANT= 1.01e-3
%
% --------------------------- THERMAL CONDUCTIVITY MODEL ----------------------%
%
% Conductivity model (CONSTANT_CONDUCTIVITY, CONSTANT_PRANDTL).
CONDUCTIVITY_MODEL= CONSTANT_PRANDTL
%
% -------------------- BOUNDARY CONDITION DEFINITION --------------------------%
%
% Navier-Stokes wall boundary marker(s) (NONE = no marker)
MARKER_HEATFLUX= ( D_BLADE, 0.0, D_SHROUD, 0.0, D_HUB, 0.0,B_BLADE, 0.0, B_SHROUD, 0.0,  B_HUB, 0.0)
%
% Periodic boundary marker(s) (NONE = no marker)
% Format: ( periodic marker, donor marker, rot_cen_x, rot_cen_y, rot_cen_z, rot_angle_x-axis, rot_angle_y-axis, rot_angle_z-axis, translation_x, translation_y, translation_z)
MARKER_PERIODIC= (D_PER1, D_PER2, 0.0, 0.0, 0.0, 0.0, 0.0, 11.25, 0.0, 0.0, 0.0,B_PER1, B_PER2, 0.0, 0.0, 0.0, 0.0, 0.0, 27.69230769230769230769230769, 0.0, 0.0, 0.0 )
%
%
%-------- INFLOW/OUTFLOW BOUNDARY CONDITION SPECIFIC FOR TURBOMACHINERY --------%
%
% Inflow and Outflow markers must be specified, for each blade (zone), following the natural groth of the machine (i.e, from the first blade to the last)
MARKER_TURBOMACHINERY= (D_INFLOW, D_OUTFLOW, B_INFLOW, B_OUTFLOW)
%
% Mixing-plane interface markers must be specified to activate the transfer of information between zones
MARKER_MIXINGPLANE_INTERFACE= (D_OUTFLOW, B_INFLOW)
%
% Giles boundary condition for inflow, outfolw and mixing-plane
% Format inlet:  ( marker, TOTAL_CONDITIONS_PT, Total Pressure , Total Temperature, Flow dir-norm, Flow dir-tang, Flow dir-span, under-relax-avg, under-relax-fourier)
% Format outlet: ( marker, STATIC_PRESSURE, Static Pressure value, -, -, -, -, under-relax-avg, under-relax-fourier)
% Format mixing-plane in and out: ( marker, MIXING_IN or MIXING_OUT, -, -, -, -, -, -, under-relax-avg, under-relax-fourier)
MARKER_GILES= (D_INFLOW, TOTAL_CONDITIONS_PT, 413.6E+03, 477.6, 1.0, 0.0, 0.0, 1.0, 0.0, D_OUTFLOW, MIXING_OUT, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.0, B_INFLOW, MIXING_IN, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.0, B_OUTFLOW, STATIC_PRESSURE_1D, 67.46E+03, 0.0, 0.0, 0.0, 0.0 , 1.0, 0.0)
%
% This option insert an extra under relaxation factor for the Giles BC at the hub and shroud levels
GILES_EXTRA_RELAXFACTOR= (0.05, 0.05)
%
%YES Non reflectivity activated, NO the Giles BC behaves as a normal 1D characteristic-based BC
SPATIAL_FOURIER= NO
%---------------------------- TURBOMACHINERY SIMULATION -----------------------------%
%
% Specify kind of architecture (AXIAL, CENTRIPETAL, CENTRIFUGAL, CENTRIPETAL_AXIAL, AXIAL_CENTRIFUGAL)
TURBOMACHINERY_KIND= CENTRIPETAL CENTRIFUGAL
%
% Specify kind of interpolation for the mixing-plane (LINEAR_INTERPOLATION, NEAREST_SPAN, MATCHING)
MIXINGPLANE_INTERFACE_KIND= LINEAR_INTERPOLATION
%
% Specify option for turbulent mixing-plane (YES, NO) default NO
TURBULENT_MIXINGPLANE= YES
%
% Specify ramp option for Outlet pressure (YES, NO) default NO
RAMP_OUTLET_PRESSURE= NO
%
% Parameters of the outlet pressure ramp (starting outlet pressure, updating-iteration-frequency, total number of iteration for the ramp)
RAMP_OUTLET_PRESSURE_COEFF= (400000.0, 10.0, 500)
%
%
% Specify Kind of average process for linearizing the Navier-Stokes equation at inflow and outflow BC included mixing-plane
% (ALGEBRAIC, AREA, MASSSFLUX, MIXEDOUT) default AREA 
AVERAGE_PROCESS_KIND= MIXEDOUT
%
% Specify Kind of average process for computing turbomachienry performance parameters
% (ALGEBRAIC, AREA, MASSSFLUX, MIXEDOUT) default AREA
PERFORMANCE_AVERAGE_PROCESS_KIND= MIXEDOUT
%Parameters of the Newton method for the MIXEDOUT average algorithm (under relaxation factor, tollerance, max number of iterations) 
MIXEDOUT_COEFF= (1.0, 1.0E-05, 15)
%
% Limit of Mach number below which the mixedout algorithm is substituted with a AREA average algorithm
AVERAGE_MACH_LIMIT= 0.05
%
%
% ------------------------ SURFACES IDENTIFICATION ----------------------------%
%
% Marker(s) of the surface in the surface flow solution file
MARKER_PLOTTING= (D_BLADE, B_BLADE)
%
%
%
% ------------- COMMON PARAMETERS DEFINING THE NUMERICAL METHOD ---------------%
%
% Numerical method for spatial gradients (GREEN_GAUSS, WEIGHTED_LEAST_SQUARES)
NUM_METHOD_GRAD= WEIGHTED_LEAST_SQUARES
%
% Courant-Friedrichs-Lewy condition of the finest grid
CFL_NUMBER= 5.0
%
% Adaptive CFL number (NO, YES)
CFL_ADAPT= YES
%
% Parameters of the adaptive CFL number (factor down, factor up, CFL min value, CFL max value )
CFL_ADAPT_PARAM= ( 0.1, 1.2, 5.0, 30.0)
%
%
% ------------------------ LINEAR SOLVER DEFINITION ---------------------------%
%
% Linear solver or smoother for implicit formulations (BCGSTAB, FGMRES, SMOOTHER)
LINEAR_SOLVER= FGMRES
%
% Preconditioner of the Krylov linear solver (ILU, LU_SGS, LINELET, JACOBI)
LINEAR_SOLVER_PREC= LU_SGS
%
% Min error of the linear solver for the implicit formulation
LINEAR_SOLVER_ERROR= 1E-4
%
% Max number of iterations of the linear solver for the implicit formulation
LINEAR_SOLVER_ITER= 10
%
%
% -------------------------- MULTIGRID PARAMETERS -----------------------------%
%
% ----------- NOT WORKING WITH PERIODIC BOUNDARY CONDITIONS !!!!! --------------%
%
%
% ----------------------- SLOPE LIMITER DEFINITION ----------------------------%
%
% Coefficient for the limiter
VENKAT_LIMITER_COEFF= 0.01
%
% Freeze the value of the limiter after a number of iterations
LIMITER_ITER= 999999
%
%
% -------------------- FLOW NUMERICAL METHOD DEFINITION -----------------------%
%
% Convective numerical method (JST, LAX-FRIEDRICH, CUSP, ROE, AUSM, HLLC,
%                              TURKEL_PREC, MSW)
CONV_NUM_METHOD_FLOW= FDS
%
% Monotonic Upwind Scheme for Conservation Laws (TVD) in the flow equations.
%           Required for 2nd order upwind schemes (NO, YES)
MUSCL_FLOW= YES
%
% Slope limiter (VENKATAKRISHNAN, VAN_ALBADA_EDGE)
SLOPE_LIMITER_FLOW= VENKATAKRISHNAN
%
% 2nd and 4th order artificial dissipation coefficients
JST_SENSOR_COEFF= ( 0.5, 0.02 )
%
% Time discretization (RUNGE-KUTTA_EXPLICIT, EULER_IMPLICIT, EULER_EXPLICIT)
TIME_DISCRE_FLOW= EULER_IMPLICIT
%
% Relaxation coefficient
%
ENTROPY_FIX_COEFF= 0.001
%
%
% -------------------- TURBULENT NUMERICAL METHOD DEFINITION ------------------%
%
% Convective numerical method (SCALAR_UPWIND)
CONV_NUM_METHOD_TURB= SCALAR_UPWIND
%
% Monotonic Upwind Scheme for Conservation Laws (TVD) in the turbulence equations.
%           Required for 2nd order upwind schemes (NO, YES)
MUSCL_TURB= NO
%
% Slope limiter (VENKATAKRISHNAN, MINMOD)
SLOPE_LIMITER_TURB= VENKATAKRISHNAN
%
% Time discretization (EULER_IMPLICIT)
TIME_DISCRE_TURB= EULER_IMPLICIT
%
% Reduction factor of the CFL coefficient in the turbulence problem
CFL_REDUCTION_TURB= 0.5
%
% Relaxation coefficient
%
%
% --------------------------- CONVERGENCE PARAMETERS --------------------------%
%
% Number of total iterations
OUTER_ITER= 300
%
% Min value of the residual (log10 of the residual)
CONV_RESIDUAL_MINVAL= -16
%
% Start convergence criteria at iteration number
CONV_STARTITER= 10
%
% Number of elements to apply the criteria
CONV_CAUCHY_ELEMS= 100
%
% Epsilon to control the series convergence
CONV_CAUCHY_EPS= 1E-6
%
%
%
% ------------------------- INPUT/OUTPUT INFORMATION --------------------------%
%
OUTPUT_FILES= (RESTART, TECPLOT_ASCII, SURFACE_TECPLOT_ASCII)
% Mesh input file
MESH_FILENAME= JQB2.su2
%
% Mesh input file format (SU2, CGNS, NETCDF_ASCII)
MESH_FORMAT= SU2
%
% Mesh output file
MESH_OUT_FILENAME= meshout.su2
%
% Restart flow input file
SOLUTION_FILENAME= restart_flow.dat
%
% Restart adjoint input file
SOLUTION_ADJ_FILENAME= solution_adj.dat
%
TABULAR_FORMAT= CSV
%
% Output file convergence history (w/o extension) 
CONV_FILENAME= history
%
% Output file restart flow
RESTART_FILENAME= restart_flow.dat
%
% Output file restart adjoint
RESTART_ADJ_FILENAME= restart_adj.dat
%
% Output file flow (w/o extension) variables
VOLUME_FILENAME= flow
%
% Output file adjoint (w/o extension) variables
VOLUME_ADJ_FILENAME= adjoint
%
% Output objective function gradient (using continuous adjoint)
GRAD_OBJFUNC_FILENAME= of_grad.dat
%
% Output file surface flow coefficient (w/o extension)
SURFACE_FILENAME= surface_flow
%
% Output file surface adjoint coefficient (w/o extension)
SURFACE_ADJ_FILENAME= surface_adjoint
%
% Writing solution file frequency
OUTPUT_WRT_FREQ= 100
%

thinks ,
huizhugeng

**Comment by NAnand-TUD** [comment]
Hi,

I think is better to run the simulation as a standard fluid problem in IncSolver as the turbo solver is exclusively for Compressible solver.

Thanks,
Nitish
